code/src/components/TaskList.js

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import moment from 'moment/moment';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { task } from 'reducers/task';
+
+export const TaskList = () => {
+  const tasklist = useSelector((store) => store.tasks.items);
+  const dispatch = useDispatch();
+
+  const onDeleteAllTasksClick = (id) => {
+    dispatch(task.actions.deleteSingleTask(id))
+  }
+  const onIsCompletedCheckboxToggle = (id) => {
+    dispatch(task.actions.toggleIfTaskIsCompleted(id))
+  }
+  const dateOptions = moment().format('ddd DD/M');
+
+  return (
+    <section>
+      <div className="header">
+        <h1>Grocery list</h1><p>{dateOptions}</p>
+      </div>
+      <ul>
+        {tasklist.map((singleTask) => {
+          return (
+            <li key={singleTask.id}>
+              <p>
+                <span>{singleTask.name}</span>
+                <label htmlFor={`task_with_id${singleTask.id}`} className="container">
+                  <input
+                    id={`task_with_id${singleTask.id}`}
```

Comment:
Really great way of giving the single item an id! Didn't see any errors using this approach.